IdentityCore/src/webview/embeddedWebview/challangeHandlers/mac/MSIDCertAuthHandler.m

```diff
@@ -72,7 +76,16 @@ + (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
                  completionHandler(NSURLSessionAuthChallengeRejectProtectionSpace, nil);
                  return;
              }
-             
+            
+            // If there is no preferred identity saved, we must set preferred identity certificate using hostname and key usage parameters
+            if (identity == NULL)
+            {
+                NSArray *arr = @[(__bridge NSString *)kSecAttrCanSign, (__bridge NSString *)kSecAttrCanEncrypt, (__bridge NSString *)kSecAttrCanDecrypt];
```

Comment:
Can you elaborate why keyUsages are set as encrypt, sign, decrypt?

Reply:
Not likely, but check if host is nil?

Reply:
Added a comment about default attributes